azure/scope/managedcontrolplane.go

```diff
@@ -364,6 +364,10 @@ func (s *ManagedControlPlaneScope) FailureDomains() []string {
 	return []string{}
 }
 
+func (s *ManagedControlPlaneScope) ManagedClusterAnnotations() map[string]string {
+	return s.ControlPlane.Annotations
+}
+
```

Comment:
nit:
Do we need this function? Can't we simply use `s.ControlPlane.Annotations` at places wherever required?

Reply:
I see why it is so. You basically implemented it as part of `ManagedMachinePoolScope` interface.

Reply:
This particular method implements `ManagedClusterScope` rather than `ManagedMachinePoolScope`. This is done in the same spirit as `ManagedClusterSpec()`.